### PR TITLE
Profiling Solution Improvement

### PR DESCRIPTION
**Pull request by stevenernst131**
Use Server Farm Service to get the correct site, since if you rely on
the respond from the diag role, it will return the resource group of the
main site, but a neighboring site on the same server farm could have a
different resource group.

### CODE REVIEW

**Comment by stevenernst131** [comment]
@puneetg1983 there may be some other improvements we want to make, when I was testing the call to get current sessions timed out, I assume because scm site was not started. Maybe we can send a warm up request earlier?

**Comment by stevenernst131** [comment]
I also removed the profiling solution from the factory so that when it starts being returned from CPU, it will not show up, since that is what you wanted.

File: app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts [hunk 1]
```diff
@@ -46,31 +46,52 @@ export class ProfilingComponent implements SolutionBaseComponent, OnInit, OnDest
     scmPath: string;
     SessionCompleted:boolean;
 
-    constructor(private _siteService: SiteService, private _daasService: DaasService, private _windowService: WindowService, private _logger: AvailabilityLoggingService) {
+    couldNotFindSite:boolean = false;
+
+    constructor(private _siteService: SiteService, private _daasService: DaasService, private _windowService: WindowService, private _logger: AvailabilityLoggingService, private serverFarmService: ServerFarmDataService) {
```
Comment (puneetg1983) [inline app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: do you want to keep the same underscore convention? Change to _serverFarmService

File: app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts [hunk 2]
```diff
@@ -46,31 +46,52 @@ export class ProfilingComponent implements SolutionBaseComponent, OnInit, OnDest
...
+            if (targetedSite) {
+                let siteName = targetedSite.name;
+                let slotName = '';
+                if (targetedSite.name.indexOf('(') >= 0) {
+                    let parts = targetedSite.name.split('(');
+                    siteName = parts[0];
+                    slotName = parts[1].replace(')', '');
+                }
+    
+                if (targetedSite) {
```
Comment (puneetg1983) [inline app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: repeated if (targetSite) ?

File: app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts [hunk 3]
```diff
@@ -46,31 +46,52 @@ export class ProfilingComponent implements SolutionBaseComponent, OnInit, OnDest
...
+
+    constructor(private _siteService: SiteService, private _daasService: DaasService, private _windowService: WindowService, private _logger: AvailabilityLoggingService, private serverFarmService: ServerFarmDataService) {
     }
 
     ngOnInit(): void {
 
         this._logger.LogSolutionDisplayed('CLR Profiling', this.data.solution.order.toString(), 'bot-sitecpuanalysis');
         
-        this.siteToBeProfiled = MetaDataHelper.getProfilingData(this.data.solution.data);         
+        this.siteToBeProfiled = MetaDataHelper.getProfilingData(this.data.solution.data);
```
Comment (puneetg1983) [inline app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: Is this really required now ? you are assigning this.siteToBeProfiled later in this function. The only possibility is when targetedSite is undefined. Could that happen ?

Comment (stevenernst131) [reply app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: removed.

File: app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts (hunk 2 above)
Comment (stevenernst131) [reply app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: removed

File: app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts (hunk 1 above)
Comment (stevenernst131) [reply app/solutions/components/specific-solutions/profiling-solution/profiling-solution.component.ts]: yup

**Pull request merged**